searchcore/src/vespa/searchcore/proton/matching/match_tools.cpp

```diff
@@ -207,12 +209,17 @@ MatchToolsFactory(QueryLimiter               & queryLimiter,
         trace.addEvent(5, "Prepare shared state for multi-threaded rank executors");
         _rankSetup.prepareSharedState(_queryEnv, _queryEnv.getObjectStore());
         _diversityParams = extractDiversityParams(_rankSetup, rankProperties);
-        DegradationParams degradationParams = extractDegradationParams(_rankSetup, rankProperties);
+        vespalib::string attribute = DegradationAttribute::lookup(rankProperties, rankSetup.getDegradationAttribute());
+        DegradationParams degradationParams = extractDegradationParams(_rankSetup, attribute, rankProperties);
+        const search::fef::FieldInfo * fieldInfo = indexEnv.getFieldByName(attribute);
```

Comment:
no need to set up range term locator if degradation is disabled

Reply:
Good point, done